api/base/parsers.py

```diff
@@ -119,3 +119,37 @@ class JSONAPIParserForRegularJSON(JSONAPIParser):
     Allows same processing as JSONAPIParser to occur for requests with application/json media type.
     """
     media_type = 'application/json'
+
+
+class JSONAPIRelationshipParser(JSONParser):
+    """
+    Parses JSON-serialized data for relationship endpoints. Overrides media_type.
+    """
+    media_type = 'application/vnd.api+json'
+
+    def parse(self, stream, media_type=None, parser_context=None):
+        res = super(JSONAPIRelationshipParser, self).parse(stream, media_type, parser_context)
+
+        if not isinstance(res, dict):
+            raise ParseError('Request body must be dictionary')
+        data = res.get('data')
+
+        if data:
+            id_ = data.get('id')
+            type_ = data.get('type')
+
+            if id_ is None:
+                raise JSONAPIException(source={'pointer': '/data/id'}, detail=NO_ID_ERROR)
+
+            if type_ is None:
+                raise JSONAPIException(source={'pointer': '/data/type'}, detail=NO_TYPE_ERROR)
+
+            return data
+
+        return {'type': None, 'id': None}
```

Comment:
I recommended this - this is because with the relationships self link you can send 
``
{
  "data": null
}
` to clear the institution.